html/includes/api_functions.inc.php

```diff
@@ -1343,6 +1445,64 @@ function list_ipsec()
     api_success($ipsec, 'ipsec');
 }
 
+
+function list_vlans()
+{
+    check_is_read();
+
+    $app      = \Slim\Slim::getInstance();
+    $router   = $app->router()->getCurrentRoute()->getParams();
+
+    $vlans       = dbFetchRows("SELECT * FROM `vlans` WHERE `vlans`.`vlan_vlan` IS NOT NULL");
+    $vlans_count = count($vlans);
+    if (!is_numeric($vlans_count)) {
+        api_error(500, 'Error retrieving VLANs');
+    }
+
+    api_success($vlans, 'vlans');
+}
+
+
+function list_ip_addresses()
```

Comment:
Should also allow to pass hostname / device_id to limit the response.

Reply:
Same as get_vlans above but with get_ip_addresses function.

Reply:
Then one function with the ability to pass hostname in would be good.